### DISCUSSION TITLE
iOS App Tracking Transparency (ATT) impact to Google Analytics That integrated alongside other analytics tools that requires ATT

### DISCUSSION THREAD

**Discussion by BakrMohamedRBM** [firebase/firebase-ios-sdk#11855] [General]
Hello,

 

I hope this post finds you well. I've been working on integrating Google Analytics SDK into my app, and I've encountered a question regarding App Tracking Transparency (ATT) on iOS and how it interacts with the SDK.

 

With the introduction of ATT, users now have the option to disable app tracking, which can impact the data collected by analytics tools. My specific concern is whether the Google Analytics SDK will continue to emit logs and collect data if a user decides to disable app tracking through ATT.

 

My Questions:

 

1- Does the Google Analytics SDK continue to emit logs and collect data when a user disables app tracking through ATT on iOS? with consideration integrations with another Analytics tools that requires ATT

 

2- Are there any specific considerations or settings within the SDK that can be adjusted to respect a user's choice regarding app tracking?

 

3- Are there any recommended workarounds or best practices for handling this situation, ensuring data privacy compliance while still obtaining useful analytics?

 

I believe it's crucial to ensure that our analytics practices align with user privacy preferences, and I want to make sure I'm following best practices with the Google Analytics SDK in this regard.

 

Any insights, guidance, or experiences you can share regarding this matter would be greatly appreciated. Thank you in advance for your assistance.

**Comment by Andro-Samurai** [comment]
>1- Does the Google Analytics SDK continue to emit logs and collect data when a user disables app tracking through ATT on iOS? with consideration integrations with another Analytics tools that requires ATT

Short answer Yes. when user asks the app not to track them, iOS will only withheld IDFA from the app
>Analytics event logging, event reporting, and conversion measurement are unaffected, but attribution is impacted if IDFA is not accessible.[ \[source\]][1]



----------


>2- Are there any specific considerations or settings within the SDK that can be adjusted to respect a user's choice regarding app tracking?

my In experience, Google always respect user privacy as a rule, so you do not have to worry if they respect the privacy or not, because they do.

> we will no longer use information (such as IDFA) that falls under ATT for the handful of our iOS apps that currently use it for advertising purposes. As such, we will not show the ATT prompt on those apps, in line with Apple’s guidance. We are working hard to understand and comply with Apple’s guidelines for all of our apps in the App Store. As our iOS apps are updated with new features or bug fixes, you’ll see updates to our app page listings that include the new App Privacy Details. ([source][2])

--------------

>3- Are there any recommended workarounds or best practices for handling this situation, ensuring data privacy compliance while still obtaining useful analytics?

Integrate SKAdNetwork for better results ([source][3])


  [1]: https://firebase.google.com/docs/ios/supporting-ios-14
  [2]: https://blog.google/products/ads-commerce/preparing-developers-and-advertisers-for-policy-updates/
  [3]: https://support.google.com/analytics/answer/13168376?hl=en

**Reply by morganchen12** [reply]
You can further customize Analytics data collection via the Consent API. See [here](https://firebase.google.com/docs/reference/swift/firebaseanalytics/api/reference/Categories/FIRAnalytics(Consent)) for reference.
